sw/device/lib/crt/crt.S

```diff
@@ -0,0 +1,134 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// CRT library
+//
+// Utility functions written in assembly that can be used before the C
+// runtime has been initialized. These functions should not be used once
+// the C runtime has been initialized.
+
+  // NOTE: The "ax" flag below is necessary to ensure that this section
+  // is allocated space in ROM by the linker.
+  .section .crt, "ax", @progbits
+
+/**
+ * Write zeros into the section bounded by the start and end pointers.
+ * The section must be word (4 byte) aligned. It is valid for the section
+ * to have a length of 0 (i.e. the start and end pointers may be equal).
+ *
+ * This function follows the standard soft float RV32 calling convention
+ * but does not require a valid stack pointer, thread pointer or global
+ * pointer.
+ *
+ * @param a0 pointer to start of section to clear (inclusive).
+ * @param a1 pointer to end of section to clear (exclusive).
```

Comment:
Please can you also document the clobbers? Maybe just a line
`Clobbers: t0`

Reply:
Done